### PR TITLE
[mono][jit] Enabling some Vector4 methods as Intrinsics on arm64.

### PR DESCRIPTION
**Pull request by LeVladIonescu**
This enables Intrinsics for the following Vector4 methods and ops : 
- Abs ()
- SquareRoot ()
- Min ()
- Max ()
- Add ()
- Subtract ()
- Multiply ()
- Divide ()
- Subtraction 
- Addition
- Division
- Multiply

Contributes to https://github.com/dotnet/runtime/issues/86957.

### CODE REVIEW

**Comment by LeVladIonescu** [comment]
Unrelated failures

**Review by jandupej — APPROVED** [review]
LGTM

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -325,6 +325,10 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 			case SN_Divide:
 			case SN_op_Division: {
 				const char *class_name = m_class_get_name (klass);
+				if (!strcmp("Vector4", class_name) && fsig->params [1]->type == MONO_TYPE_R4) {
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:328]: This new condition seems to be redundant. The following condition is sufficient enough.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -347,6 +351,14 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 			case SN_Multiply:
 			case SN_op_Multiply: {
 				const char *class_name = m_class_get_name (klass);
+				if (!strcmp("Vector4", class_name)) {
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:354]: This new condition seems to be redundant too.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:328]: Will modify it in the next PR

**Pull request merged**